### PR TITLE
updated cmake build system

### PR DESCRIPTION
**Pull request by pedrolcl**
* debug output for libsndfile, quieter output for pulseaudio
* enhanced build summary report
* revised building win32 binaries without unicode support

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -544,30 +544,37 @@ endif ( GLib2_VERSION AND GLib2_VERSION VERSION_LESS "2.26.0" )
...
+        message ( STATUS "Found libSndFile: ${SndFile_VERSION}" )
+        #cmake_print_properties(TARGETS SndFile::sndfile PROPERTIES LOCATION INTERFACE_INCLUDE_DIRECTORIES IMPORTED_CONFIGURATIONS)
+        cmake_print_variables(SndFile_WITH_EXTERNAL_LIBS SndFile_WITH_MPEG)
         list( APPEND PC_REQUIRES_PRIV "sndfile")
         if ( SndFile_WITH_EXTERNAL_LIBS )
             set ( LIBSNDFILE_HASVORBIS 1 )
         else (SndFile_WITH_EXTERNAL_LIBS)
             message ( NOTICE "Seems like libsndfile was compiled without OGG/Vorbis support." )
         endif (SndFile_WITH_EXTERNAL_LIBS)
+	else ( LIBSNDFILE_SUPPORT )
```
Comment (derselbst) [inline CMakeLists.txt:560]: Could you pls. remove the tabulator here to fix indentation?

**Review by derselbst — COMMENTED** [review]
LGTM, Pedro! Just a tiny issue below.

File: CMakeLists.txt (hunk 1 above)
Comment (pedrolcl) [reply CMakeLists.txt]: fixed

File: src/bindings/fluid_filerenderer.c [hunk 2]
```diff
@@ -289,7 +289,7 @@ new_fluid_file_renderer(fluid_synth_t *synth)
         goto error_recovery;
     }
     
-#ifdef _WIN32
+#if defined( _WIN32 ) && defined( _UNICODE )
```
Comment (pedrolcl) [inline src/bindings/fluid_filerenderer.c:292]: @derselbst : I think that the #if is wrongly placed here. If should be placed just before the `else`.

**Comment by pedrolcl** [comment]
@carlo-bramini : this branch may be used for debugging the issues you have found with libsndfile. I couldn't reproduce your findings, but added some debug output to the cmake scripts that maybe you find useful for diagnosing the issues.

**Review by derselbst — APPROVED** [review]

**Pull request merged**